cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -250,7 +251,7 @@ private void addProgram(String phaseName, WorkflowProgramAdder programAdder) {
       String connectorName = connectorInfo.getName();
       String datasetName = connectorDatasets.get(connectorName);
       if (datasetName == null) {
-        datasetName = UUID.randomUUID().toString();
+        datasetName = Integer.toString(connectorNum++);
```

Comment:
anything shorter than that? I know nothing's going to fix this issue completely, but trying to reduce the impact.

``"conn" + connectorNum++``, 
``"localds" + connectorNum++``?